Problem run with --release --no-track-widget-creation flags, android studio

hi guys,
when run with `--release and --no-track-widget-creation flags` i have this result:
 
```
Initializing gradle...
Resolving dependencies...
Launching lib\main.dart on LG G2 in release mode...
Gradle task 'assembleRelease'...

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForRelease'.
> java.lang.IllegalStateException: Expected BEGIN_ARRAY but was STRING at line 1 column 1 path $

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

_Originally posted by @Gyorgio88 in https://github.com/flutter/flutter/issues/20260#issuecomment-440898843_

--- a/packages/flutter_tools/bin/xcode_backend.sh
+++ b/packages/flutter_tools/bin/xcode_backend.sh
@@ -183,8 +183,7 @@ BuildApp() {
       --${build_mode}                                                       \
       --ios-arch="${archs}"                                                 \
       ${flutter_engine_flag}                                                \
-      ${local_engine_flag}                                                  \
-      ${track_widget_creation_flag}
+      ${local_engine_flag}
 
     if [[ $? -ne 0 ]]; then
       EchoError "Failed to build ${project_path}."

--- a/packages/flutter_tools/lib/src/commands/build_aot.dart
+++ b/packages/flutter_tools/lib/src/commands/build_aot.dart
@@ -38,6 +38,10 @@ class BuildAotCommand extends BuildSubCommand with TargetPlatformBasedDevelopmen
         defaultsTo: false,
         help: 'Report timing information about build steps in machine readable form,',
       )
+      ..addFlag('track-widget-creation',
+        defaultsTo: false,
+        hide: true,
+      )
       ..addMultiOption('ios-arch',
         splitCommas: true,
         defaultsTo: defaultIOSArchs.map<String>(getNameForIOSArch),

--- a/packages/flutter_tools/lib/src/commands/build_aot.dart
+++ b/packages/flutter_tools/lib/src/commands/build_aot.dart
@@ -38,6 +38,8 @@ class BuildAotCommand extends BuildSubCommand with TargetPlatformBasedDevelopmen
         defaultsTo: false,
         help: 'Report timing information about build steps in machine readable form,',
       )
+      // track-widget-creation is exposed as a flag here but ignored to deal with build
+      // invalidation issues - there are no plans to support it for AOT mode.
       ..addFlag('track-widget-creation',
         defaultsTo: false,
         hide: true,

